### ISSUE TITLE
We detected that the Chromium Renderer process just crashed Cypress CI error

### ISSUE THREAD

**Issue by HeshamMoneer** [cypress-io/cypress#24905]
### Current behavior

I get this error when I run my Cypress test scripts on Azure CI server.
```
<--- Last few GCs --->

[8736:0x2f6c003b0000]   987167 ms: Scavenge 397.7 (437.8) -> 391.8 (438.0) MB, 6.0 / 0.0 ms  (average mu = 0.998, current mu = 0.998) allocation failure; 
[8736:0x2f6c003b0000]   988392 ms: Scavenge 397.9 (438.0) -> 392.6 (438.8) MB, 8.7 / 0.0 ms  (average mu = 0.998, current mu = 0.998) allocation failure; 
[8736:0x2f6c003b0000]   988595 ms: Scavenge 399.3 (438.8) -> 392.6 (439.8) MB, 10.2 / 0.0 ms  (average mu = 0.998, current mu = 0.998) allocation failure; 


<--- JS stacktrace --->

[8736:1124/101216.876419:ERROR:v8_initializer.cc(730)] V8 process OOM (Oilpan: Reserving memory.).

We detected that the Chromium Renderer process just crashed.

This is the equivalent to seeing the 'sad face' when Chrome dies.

This can happen for a number of different reasons:

- You wrote an endless loop and you must fix your own code
- You are running Docker (there is an easy fix for this: see link below)
- You are running lots of tests on a memory intense application
- You are running in a memory starved VM environment
- There are problems with your GPU / GPU drivers
- There are browser bugs in Chromium
```
I tried adding the option **--browser chrome** to the **cypress run** command, but it makes the tests fail many assertions (assertions that would not fail if I run without the browser chrome option) unexpectedly. 

PS. the same test scripts would run fine in cypress 10.7.0, electron 102. But now, they fail with the mentioned error in cypress 11.2.0, electron 106. I think the failure is more related to electron browser version, but I am not certain.

### Test code to reproduce

npx cypress run

### Cypress Version

11.2.0

### Node version

12.22.9

### Operating System

Ubuntu 22.04.1 LTS

**Comment by HeshamMoneer** [comment]
@nagash77 I noticed that I had such a failing run in Cypress 10. The difference I think (which might have caused the error) is the electron browser version. In the successful runs, the electron browser version was 102, but in the failing runs, the version was 106.

**Comment by HeshamMoneer** [comment]
I still have an issue, running the scripts headed in chrome works fine, but running them headlessly has a lot of assertions failing unjustifiably. The same assertions that would pass in electron or headed chrome runs, would fail in headless chrome runs.

**Comment by NeeluSharmaVeersa** [comment]
I also have same issue, running the scripts headed /headless in chrome or electron also crashed after some time
Please suggest if this is due to Browser version or cypress itself

Chrome Version - 107
Electron Version - 100
Cypress - 9.7.0

**Comment by alesiaviktorchyk** [comment]
We also are getting `We detected that the Chromium Renderer process just crashed` almost every time with Cypress 11.2.0. Though everything was ok with Cypress 9.6.0.

We are using cypress circleci-orb (version 2.2.0) and `cypress/browsers:node16.16.0-chrome107-ff107` executor.

**Comment by Jacek-fstack** [comment]
I keep getting this in my CI - GH actions; when running locally it sometimes pops up as well

**Comment by mi4i0** [comment]
Also started to be experienced in a log of chrome crashes in headless mode.
It started to appear after moving from 8.4.0 to Cypress 10.11.0 (moved with one-time migration without migrating to v.9...)
We are running our tests in parallel on machines with 2 cores and 4 Gb RAM
SemaphoreCI as CI service
NodeJS 18.4
Chrome

**Comment by AtofStryker** [comment]
We are releasing in `12.4.0` next week an option called `experimentalMemoryManagement`, which should help cut down on this issue. Would any of you be willing to try this in CI and see if the issue improves?

**Comment by Jacek-fstack** [comment]
@AtofStryker we'll try it for sure

**Comment by ClaasWilke** [comment]
> We are releasing in `12.4.0` next week an option called `experimentalMemoryManagement`, which should help cut down on this issue. Would any of you be willing to try this in CI and see if the issue improves?

For me, the new `experimentalMemoryManagement` works fine. Our OOM-Errors are gone :)

**Comment by mschile** [comment]
Closing as a duplicate of https://github.com/cypress-io/cypress/issues/23391. Please try out the new [experimentalMemoryManagement](https://docs.cypress.io/guides/references/experiments) flag available in Cypress 12.4.0. There is also a memory improvement in Cypress 12.5.0. If you are still experiencing issues after upgrading, please log a new [memory issue](https://github.com/cypress-io/cypress/issues/new?assignees=&labels=&template=2-memory-issue.yml).

**Comment by rakeshnambiar** [comment]
Hi, I am using the latest version of Cypress which is `12.14.0` and I also have ` experimentalMemoryManagement: true,` but still it's crashed.

```
We detected that the Chromium Renderer process just crashed.

This can happen for a number of different reasons.

If you're running lots of tests on a memory intense application.
  - Try increasing the CPU/memory on the machine you're running on.
  - Try enabling experimentalMemoryManagement in your config file.
  - Try lowering numTestsKeptInMemory in your config file during 'cypress open'.
  ```
I am now also planning to try with `numTestsKeptInMemory: 20,`

**Comment by AnushaAshokReddy** [comment]
I still have this issue happening in jenkins. 
We are running only 10 tests in parallel with 5 containers and 2 of them fail with the chrome renderer issue. 
Tried the 'experimentalMemoryManagement' flag 'true' - this also did not solve the issue. 

Here is the screenshot : 
<img width="527" alt="image" src="https://github.com/cypress-io/cypress/assets/44580961/a765043c-355a-4a57-a0f1-360173cfb60a">

**Comment by mschile** [comment]
@AnushaAshokReddy, it looks like you're using an older version of Cypress which does not support `experimentalMemoryManagement`. Please see this [comment](https://github.com/cypress-io/cypress/issues/24905#issuecomment-1411118697) for required Cypress versions.

**Comment by AnushaAshokReddy** [comment]
@mschile , I am using the latest version 12.17.0

**Comment by mschile** [comment]
@AnushaAshokReddy, the latest version of Cypress doesn't have the [text](https://github.com/cypress-io/cypress/blob/v12.17.0/packages/errors/src/errors.ts#L629-L640) that is displayed in your screenshot. Is the screenshot from a different version?

**Comment by scorpyto** [comment]
I have the same issue. I run one spec files which imports other spec files with multiple tests (this approach my be wrong but it is working to handle all spec files into only 1)

**Comment by AnushaAshokReddy** [comment]
@mschile , thank you, 
fixed it by updating the version (12.17.0) in jenkins file and by adding the following : 
numTestsKeptInMemory: 0,
experimentalMemoryManagement: true,

**Comment by dummulokesh2105** [comment]
am using the cypress version 12.7.4; Browser - Electron 106(headless); Node version v16.20.1 and the same below in m projectory and still am getting this issue. Can anyone help me here please

numTestsKeptInMemory: 0,
experimentalMemoryManagement: true,

Below is the error :

We detected that the Chromium Renderer process just crashed.
This can happen for a number of different reasons.
If you're running lots of tests on a memory intense application.
  - Try increasing the CPU/memory on the machine you're running on.
  - Try enabling experimentalMemoryManagement in your config file.
  - Try lowering numTestsKeptInMemory in your config file during 'cypress open'.

**Comment by akshaygupta5233** [comment]
Hi Team, 
Please help us on this!

**Comment by sosnowskijakub** [comment]
Hi! 
I still have the same issue locally and in azure devops, 
cypress 13.7.3
node 20.8.0

numTestsKeptInMemory: 0,
experimentalMemoryManagement: true,

Below is the error :
<img width="758" alt="Screenshot 2024-04-19 at 13 03 54" src="https://github.com/cypress-io/cypress/assets/48626087/6c371932-dfb7-4749-9a75-ec05816f4a18">

**Comment by sosnowskijakub** [comment]
The only walkaround I found is using CYPRESS_NO_COMMAND_LOG=1
but it's not the easiest way to work with cypress

**Comment by mbhagchandani** [comment]
hi, we were facing the same issue. It was because we introduced a new endpoint in the backend that would be called when the app loads for the first time, but we did not mock the response to be expected in cypress. Cypress would wait for the response for a long time before moving forward and we started getting this error in our tests. 
Hope this helps others

**Comment by singh-sukhpreet** [comment]
we were also facing the same issue and we fix it by changing the chrome version from latest to stable in our github actions pipeline
